servicetalk-concurrent-jdkflow/src/test/java/io/servicetalk/concurrent/jdkflow/JdkFlowAdaptersTest.java

```diff
@@ -0,0 +1,190 @@
+        verify(subscription).cancel();
+    }
+
+    @Test
+    public void toFlowSuccess() {
+        verifyFlowSuccess(toFlowPublisherAndSubscribe(from(1)));
+    }
+
+    @Test
+    public void toFlowFromSourceSuccess() {
+        PublisherSource<Integer> source = s -> s.onSubscribe(new ScalarValueSubscription<>(1, s));
+        Subscriber<Integer> subscriber = toFlowPublisherFromSourceAndSubscribe(source);
+        verifyFlowSuccess(subscriber);
+    }
+
+    private void verifyFlowSuccess(final Subscriber<Integer> subscriber) {
+        verify(subscriber).onSubscribe(any());
+        verify(subscriber).onNext(1);
+        verify(subscriber).onComplete();
+        verifyNoMoreInteractions(subscriber);
+    }
+
+    @Test
+    public void toFlowError() {
+        verifyFlowError(toFlowPublisherAndSubscribe(failed(DELIBERATE_EXCEPTION)));
+    }
+
+    @Test
+    public void toFlowFromSourceError() {
+        PublisherSource<Integer> source = s -> {
+            s.onSubscribe(EMPTY_SUBSCRIPTION);
+            s.onError(DELIBERATE_EXCEPTION);
+        };
+        verifyFlowError(toFlowPublisherFromSourceAndSubscribe(source));
+    }
+
+    private void verifyFlowError(final Subscriber<Integer> subscriber) {
+        verify(subscriber).onSubscribe(any());
+        verify(subscriber).onError(DELIBERATE_EXCEPTION);
+        verifyNoMoreInteractions(subscriber);
+    }
+
+    @Test
+    public void toFlowCancel() {
+        TestPublisher<Integer> stPublisher = new TestPublisher<>();
+        Subscriber<Integer> subscriber = toFlowPublisherAndSubscribe(stPublisher);
+        TestSubscription subscription = new TestSubscription();
+        stPublisher.onSubscribe(subscription);
+        assertThat("Source not subscribed.", stPublisher.isSubscribed(), is(true));
+        ArgumentCaptor<Subscription> subscriptionCaptor = ArgumentCaptor.forClass(Subscription.class);
+        verify(subscriber).onSubscribe(subscriptionCaptor.capture());
+        subscriptionCaptor.getValue().cancel();
+        assertThat("Subscription not cancelled.", subscription.isCancelled(), is(true));
+    }
+
+    @Test
+    public void toFlowFromSourceCancel() {
+        PublisherSource.Subscription srcSubscription = mock(PublisherSource.Subscription.class);
+        PublisherSource<Integer> source = s -> s.onSubscribe(srcSubscription);
+        Subscriber<Integer> subscriber = toFlowPublisherFromSourceAndSubscribe(source);
+        ArgumentCaptor<Subscription> flowSubscriptionCaptor = ArgumentCaptor.forClass(Subscription.class);
+        verify(subscriber).onSubscribe(flowSubscriptionCaptor.capture());
+        flowSubscriptionCaptor.getValue().cancel();
+        verify(srcSubscription).cancel();
+    }
+
+    private Subscriber<Integer> toFlowPublisherAndSubscribe(
+            final io.servicetalk.concurrent.api.Publisher<Integer> stPublisher) {
+        Publisher<Integer> flowPublisher = JdkFlowAdapters.toFlowPublisher(stPublisher);
```

Comment:
nit: Use static import for `JdkFlowAdapters.toFlowPublisher`

Reply:
sure